server.js

```diff
@@ -1,20 +1,143 @@
+  response.cookie("access", accessToken, {
+    maxAge: (expiresIn - 60) * 1000,
+  });
+  response.cookie("refresh", refreshToken);
+  response.redirect("/vote");
+});
+
+app.get("/oauth/spotify/logout", async (request, response) => {
+  response.clearCookie("access");
+  response.clearCookie("refresh");
+  response.status(200).send("Cookies cleared");
+});
+
+app.get("/oauth/spotify/refreshtoken", async (request, response) => {
+  const refreshToken = request.cookies.refresh;
+  try {
+    const {
+      access_token: newAccessToken,
+      expires_in: expiresIn,
+      refresh_token: newRefreshToken,
+    } = await refreshUserAccessToken(refreshToken);
+    response.cookie("access", newAccessToken, {
+      maxAge: (expiresIn - 60) * 1000,
+    });
+    if (newRefreshToken) {
+      response.cookie("refresh", newRefreshToken);
+    }
+  } catch (error) {
+    response.redirect("/oauth/spotify/authorize");
+    console.error(error);
+  }
+});
+
+// Client Credentials Flow
+
+app.get("/oauth/spotify/apptoken", async (request, response) => {
+  const {
+    access_token: appToken,
+    expires_in: expiresIn,
+  } = await getAppAccessToken();
+  response.cookie("apptoken", appToken, {
+    maxAge: (expiresIn - 60) * 1000,
+  });
+  response.status(200).send("App Token Cookie is set!");
+});
+
+// Spotify API
+
+app.get("/api/user/profile", async (request, response) => {
+  const accessToken = request.cookies.access;
+  const refreshToken = request.cookies.refresh;
+  if (!accessToken && !refreshToken) {
+    response.redirect("/oauth/spotify/authorize");
+    return;
+  }
+  if (!accessToken) {
+    const {
+      access_token: newAccessToken,
+      expires_in: expiresIn,
+      refresh_token: newRefreshToken,
+    } = await refreshUserAccessToken(refreshToken);
+
+    response.cookie("access", newAccessToken, {
+      maxAge: (expiresIn - 60) * 1000,
+    });
+    if (newRefreshToken) {
+      response.cookie("refresh", newRefreshToken);
+    }
+    const userProfileData = await getCurrentUserProfile(newAccessToken);
+    response.send(userProfileData);
+    return;
+  }
+  try {
+    const userProfileData = await getCurrentUserProfile(accessToken);
+    response.send(userProfileData);
+  } catch (error) {
+    console.error(error);
+  }
 });
 
 app.get("/api/episode/:id", async (request, response) => {
   const { id } = request.params;
-  const token = request.headers.authorization;
-  const episodeData = await getEpisodeInfo(token, id);
+  const accessToken = request.cookies.apptoken;
+  if (!accessToken) {
+    const {
+      access_token: newAppToken,
+      expires_in: expiresIn,
+    } = await getAppAccessToken();
+    response.cookie("apptoken", newAppToken, {
+      maxAge: (expiresIn - 60) * 1000,
+    });
+    const episodeData = await getEpisodeInfo(newAppToken, id);
+    response.send(episodeData);
+    return;
+  }
+  const episodeData = await getEpisodeInfo(accessToken, id);
   response.send(episodeData);
```

Comment:
double of code, replace it with one function , which will do that, or think about using it once with a changing const

```suggestion
    const episodeData = await getEpisodeInfo(newAppToken, id);
    response.send(episodeData);
    return;
  }
  const episodeData = await getEpisodeInfo(accessToken, id);
  response.send(episodeData);
```

something like that 

`````js
const episodeData = await getEpisodeInfo(newAppToken || accessToken, id);
  response.send(episodeData);

`````